doc/manual/source/user_guide/InSituPythonAnalysis.rst

```diff
@@ -0,0 +1,269 @@
+.. _in_situ_python_analysis:
+
+In Situ Python Analysis
+=======================
+
+This feature enables Enzo to do in situ analysis using Python, by calling and using Python during simulation runtime.
+This avoids outputting data to disk before conducting analysis.
+Here, we use `yt <https://yt-project.org>`__ as an analysis platform.
+
+Requirements
+------------
+
+Below are links to the build and runtime requirements, which must be installed.
+We can compile ``libyt`` using different options based on our used cases, so that Enzo can have different in situ analysis feature when it links to ``libyt``.
+
+A brief description of each mode (option) is shown here. The options are mutually independent. Please follow the instructions in ``libyt`` documents:
+
+* `libyt`_: a C shared library for in situ analysis.
+
+  * **Serial Mode** (``-DSERIAL_MODE=ON``): Compile ``libyt`` using GCC compiler.
+
+  * **Parallel Mode** (``-DSERIAL_MODE=OFF``): Compile ``libyt`` using MPI compiler and use it in parallel computation.
+
+  * **Normal Mode** (``-DINTERACTIVE_MODE=OFF``): Shut down and terminate all the processes including simulation, if there are errors during in situ analysis using Python.
+
+  * **Interactive Mode** (``-DINTERACTIVE_MODE=ON``): Fault-tolerant to Python and support interactive Python prompt and reloading script feature.
+
+  * **Jupyter Kernel Mode** (``-DJUPYTER_KERNEL=ON``): Fault-tolerant to Python and support using Jupyter Notebook / JupyterLab UI to connect to simulation and do in situ analysis.
+
```

Comment:
I would clarify that these are compilation options for libyt and not for Enzo

Reply:
This is fixed.